src/dashboard/utilities.py

```diff
@@ -46,3 +47,72 @@ def toggle_classname(class_str: str, class_to_toggle: str) -> str:
         return set_classname(class_str, class_to_toggle, False)
     else:
         return set_classname(class_str, class_to_toggle, True)
+
+
+def pluralize(unit: str, amount: int) -> str:
+    """Pluralize a unit if the amount is greater than 1.
+
+    Args:
+        unit (str): The unit.
+        amount (int): The amount.
+
+    Returns:
+        str: The unit with s appended
+        if amount is greater than 1,
+        otherwise the unit.
+    """
+    if amount == 1:
+        return unit
+
+    return f"{unit}s"
+
+
+def singularize(article: str, amount: int) -> str:
+    """Singularize an amount if it is equal to one.
+
+    Args:
+        article (str): The indefinite
+        article of an amount of something.
+        E.g. "A" or "An" in english.
+        amount (int): An amount.
+
+    Returns:
+        str: `article` if the amount
+        is equal to 1, otherwise
+        a string of amount.
+    """
+    if amount == 1:
+        return article
+
+    return str(amount)
+
+
+def to_human_time_delta(t1: datetime, t2: datetime) -> str:
```

Comment:
Is there a particular reason passing two timestamps is preferable to passing a timedelta?

Reply:
In this implementation it is used to display the formatted date of the oldest time. Although using timedelta as the parameter and calculating the date by calling `datetime.now()` in `to_human_time_delta()` could work, it would prevent the ability to pass in arbitrary times as well as not guarantee consistent results when calling the function multiple times.